HH_custom_primary_key.md

```diff
@@ -0,0 +1,97 @@
+###Example: Create a model with custom primary key `name:string` instead of autogenerated `id:integer`
+
+IMPORTANT NOTE: this example was chosen for simplicity.  It is not advisable to use something that is likely to change
+in the future as primary key.  
+
+Create project named chat:
+
+```console
+$mix phoenix.new chat
+$cd chat
+$mix ecto.create
+```
+
+To create User model run: 
+```console
+$mix phoenix.gen.json User users name:string age:integer
+```
+
+Modify 4 generated files:
+
+**1. Modify models/user.ex like so:**
+
+* Add `@primary_key {:name, :string, []}` right above `schema "users" do` and remove `field :name` from below it:
+```elixir
+@primary_key {:name, :string, []}  # ADDED THIS LINE
+schema "users" do
+  #field :name, :string  REMOVED THIS LINE
+  field :age, :integer
+
+  timestamps
+end
+```
+    
+* Add below code somewhere in this file to be able to specify name instead of id in the URL.
+    Based on this example https://github.com/gjaldon/phoenix_params
+```elixir
+defimpl Phoenix.Param, for: Chat.User do 
+    def to_param(user) do
+        user.name
+   end
+end
+```
+And below is another modification we need to be able to specify name instead of id in the URL
+(again, based on https://github.com/gjaldon/phoenix_params)
+
+**2. Modify web/controllers/user_controller.ex show, update, and delete methods**
+
+NOTE the use of `Repo.get_by` instead of `Repo.get` and `name: name` instead of `id`
```

Comment:
Actually, you don't need to to those changes. `Repo.get!` will always use the primary key. So as long as you change the primary key, you are fine. :)